app/code/Magento/Eav/Model/Attribute/Data/File.php

```diff
@@ -146,7 +146,7 @@ protected function _validateByRules($value)
             return $this->_fileValidator->getMessages();
         }
 
-        if (!empty($value['tmp_name']) && !is_uploaded_file($value['tmp_name'])) {
+        if (empty($value['tmp_name']) && !is_uploaded_file($value['tmp_name'])) {
```

Comment:
Hi @Nazar65 the condition `empty($value['tmp_name']) && !is_uploaded_file($value['tmp_name'])` will evaluate to:
 - `false` is  `$value['tmp_name']` is set
 - `true`if `$value['tmp_name']` is empty
The second part of the condition will be checked only if `$value['tmp_name']` is empty, that does not make sence in this case.

That's why I suggest changing this condition to fail validation in both cases if `tmp_name` is empty or file is not uploaded
```suggestion
        if (empty($value['tmp_name']) || !is_uploaded_file($value['tmp_name'])) {
```

Reply:
Ohh. :+1:  thanks, now clearly.